code/game/objects/items/robot/robot_items.dm

```diff
@@ -448,7 +457,7 @@
 		if(R.emagged)
 			hitdamage = emaggedhitdamage
 	switch(mode)
-		if(DISPENSE_LOLLIPOP_MODE)
+		if(DISPENSE_LOLLIPOP_MODE, DISPENSE_ICECREAM_MODE)

```

Comment:
because you catch it up here

Reply:
Different procs, dude. That's the afterattack, other one is attack_self.